romanesco/plugins/gaia/classes.py

```diff
@@ -0,0 +1,98 @@
+import os
+import romanesco
+from romanesco.specs import Port, PortList
+from romanesco.plugins.gaia import PACKAGE_DIR as GAIA_DIR
+
+
+class Task(object):
+    def __init__(self, *args, **kwargs):
+        self.inputs = {}
+        self.mode = kwargs.get("mode", "python")
+
+        for pos, port in enumerate(self.__inputs__):
+            self.inputs[port.name] = dict(port.items() + [("data", args[pos])])
+
+    @property
+    def script(self):
+        raise Exception("Must be implemented in subclass!")
+
+    def parse(self):
+        return {
+            "inputs": self.__inputs__.values(),
+            "outputs": self.__outputs__.values(),
+            "script": self.script,
+            "mode": self.mode}
+
+    def run(self):
+        raw = romanesco.run(self.parse(), inputs=self.inputs)
+        ret = []
+
+        for pos, port in enumerate(self.__outputs__):
+            try:
+                ret.append(raw[port.name]['data'])
+            except KeyError:
+                # Error here?
+                pass
+
+        return ret[0] if len(ret) == 1 else tuple(ret)
```

Comment:
This ```Task``` object should probably be the task object from the spec.  I would like to see the two merged but only if we decide that that this is how we want to move forward (especially with class-level variables that define the ports on an analysis).